app/questionnaire/rules.py

```diff
@@ -281,3 +290,34 @@ def get_list_count(list_store, list_name):
 
 def is_goto_rule(rule):
     return any(key in rule.get('goto', {}) for key in ('when', 'block', 'group'))
+
+
+def get_default_answer(answer_id, answer_store, schema):
+    try:
+        block = schema.get_block_for_answer_id(answer_id)
+    except TypeError:
+        raise NoDefaultAnswerError
+
+    question = None
+    try:
+        question = block['question']
+    except KeyError:
+        variants = block['question_variants']
+        for variant in variants:
```

Comment:
We have a method in schema_utils.py to evaluate variants. `choose_question_to_display` I believe, though that could be named better :)

Reply:
Using that should eliminate all of this try/except

Reply:
You will need to pass in some more arguments, but those should be there anyway, as currently this won't work for rules based on metadata / lists.

Reply:
Just spoken to @ajmaddaford about this. I think we will make a change to the schema validations so that question variants in the same block require the same default answer. Won't need to check which of the variants was displayed then.